### PR TITLE
BUG: Series.replace with to_replace dict and value=None, nonsensical error message

### LINKED ISSUE #46004: BUG: Series.replace with to_replace dict and value=None
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
s = pd.Series([1,2,3,4])
s.replace(to_replace={3:33}, value=None)
```

### Issue Description

```python
import pandas as pd
s = pd.Series([1,2,3,4])
s.replace(to_replace={3:33}, value=7)
```
yields the error as expected:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/drams/anaconda3/envs/plombier-dev/lib/python3.10/site-packages/pandas/core/series.py", line 4960, in replace
    return super().replace(
  File "/Users/drams/anaconda3/envs/plombier-dev/lib/python3.10/site-packages/pandas/core/generic.py", line 6685, in replace
    raise ValueError(
ValueError: Series.replace cannot use dict-like to_replace and non-None value
```
However, 
```python
import pandas as pd
s = pd.Series([1,2,3,4])
s.replace(to_replace={3:33}, value=None)
```
yields exactly the same error

### Expected Behavior

I suspect this change occurred because we now treat `None` in the value field explicitly.  However, the error message makes no sense anymore.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 06d230151e6f18fdb8139d09abf539867a8cd481
python           : 3.10.2.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.1
numpy            : 1.22.2
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 22.0.3
setuptools       : 60.9.1
Cython           : None
pytest           : 7.0.1
hypothesis       : 6.36.2
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : 2022.01.0
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : 2.8.0
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.01.0
scipy            : None
sqlalchemy       : 1.4.31
tables           : 3.7.0
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
</details>

### PR DESCRIPTION
**Pull request by Kyrpel**
- [x] closes #46004
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/methods/test_replace.py [hunk 1]
```diff
@@ -557,6 +557,14 @@ def test_pandas_replace_na(self):
         exp = pd.Series(["CC", "CC", "CC-REPL", "DD", "CC", "", pd.NA], dtype="string")
         tm.assert_series_equal(result, exp)
 
+    def test_replace_dictlike_value_None(self):
+        # GH46004
+        ser = pd.Series([1, 2, 3, 4])
+        result = ser.replace(to_replace={3: 33}, value=None)
```
Comment (phofl) [inline pandas/tests/series/methods/test_replace.py:563]: I don't think this is supposed to work. We are using lib.no_default as a sentinel to check if value was provided.

**Review by phofl — CHANGES_REQUESTED** [review]
We can either change the default for value (but I think lib.no_default was added on purpose) or clarify the error message

File: pandas/tests/series/methods/test_replace.py (hunk 1 above)
Comment (Kyrpel) [reply pandas/tests/series/methods/test_replace.py:563]: It seems like when we have explicitly to_replace=dict_like  and value=None it interpreted as a special case where None is different from lib.no_default. Maybe we need another conditional statement somewhere.

**Comment by denizn** [comment]
> * [x]  closes [BUG: Series.replace with to_replace dict and value=None, nonsensical error message #46004](https://github.com/pandas-dev/pandas/issues/46004)
> * [x]  [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
> * [x]  All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
> * [ ]  Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Hi there, just curious if this issue still persist. I seem to be getting the same error with series.replace. Could anyone enlighten me if there is a version that would have resolved this?

Thank you!

**Pull request closed without merging**